test/plugin/base_test.rb

```diff
@@ -2226,9 +2288,36 @@ def verify_cascading_json_detection_with_log_entry_fields(
         d.run
       end
       verify_log_entries(1, COMPUTE_PARAMS, 'jsonPayload') do |entry|
-        assert_equal expected_value, entry[log_entry_field],
-                     "Index #{index} failed. #{expected_value} is expected" \
-                     " for #{log_entry_field} field."
+        if subfields_with_default
```

Comment:
This is getting really complex, and very likely doesn't belong in the base test implementation. Normally, we'd leave a TODO and open a bug to refactor later so as not to hold up this PR, but I'll let you decide what the right course of action is here.